bayesian/utility/v2/ca_response_builder.py

```diff
@@ -18,11 +18,51 @@
 from urllib.parse import quote
 import logging
 from bayesian.utility.db_gateway import GraphAnalyses
-from bayesian.utils import version_info_tuple, convert_version_to_proper_semantic
+from bayesian.utils import version_info_tuple, convert_version_to_proper_semantic, \
+    server_create_analysis
+from typing import Dict, List, Tuple, Union, Set
+import re
+from collections import namedtuple
+from abc import ABC
+from flask import g
+
 
 logger = logging.getLogger(__name__)
 
 
+def validate_version(version: str) -> bool:
+    """Version should not contain special Characters."""
+    if re.findall('[!@#$%^&*()]', version):
+        return False
+    return True
+
+
+def unknown_package_flow(ecosystem: str, unknown_pkgs: Set[namedtuple], api_flow: bool) -> bool:
+    """Unknown Package flow."""
+    for pkg in unknown_pkgs:
+        # Enter the unknown path: Trigger bayesianApiFlow
+        server_create_analysis(ecosystem, pkg.name, pkg.version, user_profile=g.decoded_token,
+                               api_flow=api_flow, force=False, force_graph_sync=True)
+    return True
+
+
+class NormalizedPackages:
+    """Duplicate free Package List."""
+
+    def __init__(self, packages: List):
+        """Create NormalizedPackages by removing all duplicates from packages."""
+        self.packages = packages
+
+    @property
+    def all_packages(self) -> List[namedtuple]:
+        """All Packages in Tuple format."""
+        Package = namedtuple("Package", ["name", "version"])
+        pkg_list = [Package(name=pkg['name'],
+                            version=pkg['version'])
+                    for pkg in self.packages]
+        return pkg_list
```

Comment:
So the purpose of this class is to transform the List of package_versions dict to List of package_version namedtuple , so that it becomes hashable and usable with Set? In that case I would recommend define it is a function because class is not a good type for this problem.
```suggestion
Package = namedtuple("Package", ["name", "version"])
def normlize_packages(packages: List[Dict]) -> List[Package]:
    return [Package(p['name'], p['version']) for p in packages]
```

Reply:
Ack!